locale/en/docs/guides/diagnostics-flamegraph.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Diagnostics - Flame Graphs
+layout: docs.hbs
+---
+
+# Flame Graphs
+
+## What's a flame graph useful for?
+
+Flame graphs are a way of visualizing CPU time spent in functions. They can help you pin down where you spend too much time doing synchronous operations.
+
+## How to create a flame graph
+
+You might have heard creating a flame graph for Node.js is difficult, but that's not true (anymore).
+Solaris vms are no longer needed for flame graphs on linux!
+
+1. install lldb in your system (eg. with apt-get)
+1. install llnode according to readme in [github project](https://github.com/nodejs/llnode)
+1. install linux-tools-common (eg. with apt-get)
+1. try running `perf` - it might complain about missing kernel modules, install them too
+1. install stackvis `npm i -g stackvis`
+1. run node with perf enabled `perf record -e cycles:u -g -- node --perf-basic-prof app.js`
+1. disregard warnings unless they're saying you can't run perf due to missing packages; you may get some warnings about not being able to access kernel module samples which you're not after anyway.
+1. run `perf script`, but pipe it through some cleanup: `perf script | egrep -v "( __libc_start| LazyCompile | v8::internal::| Builtin:| Stub:| LoadIC:|\[unknown\]| LoadPolymorphicIC:)" | sed 's/ LazyCompile:[*~]\?/ /' > perfs.out` [explanation](#about-perf-script-filtering)
+1. run `stackvis perf < perfs.out > flamegraph.htm`
+
+Now open the flame graph file in your favorite browser and watch it burn. It's color-coded so you can focus on the most saturated orange bars first. They're likely to represent CPU heavy functions.
+
+Worth mentioning - if you click an element of a flame graph a zoom-in of its surroundings will get displayed above the graph.
+
+### Use perf to sample a running process
+
+```
+perf record -F99 -p `pgrep -n node` -g -- sleep 3
+```
+
+Wait, what is that `sleep 3` for? It's there to keep the perf running - despite `-p` option pointing to a different pid, the command needs to be executed on a process and end with it. Just a convenient way to keep perf going for a given time.
+
+Why is `-F` (profiling frequency) set to 99? I honestly don't know, Netflix guys used that value. Results look good. You can adjust if you want.
```

Comment:
Too honest. This part will have to change. On the other hand, I think we should provide a reasonable default here instead of a lengthy explanation of what -F really does and how to decide on a value.